src/components/Species/index.tsx

```diff
@@ -422,14 +434,18 @@ export default function SpeciesList({ reloadData, species }: SpeciesListProps):
     snackbar.toastError(snackbarMessage);
   };
 
-  const OnEditSpecies = () => {
-    setSelectedSpecies(selectedSpeciesRows[0]);
-    setEditSpeciesModalOpen(true);
+  const openEditSpeciesModal = async (speciesId: number) => {
+    const speciesResponse = await getSpecies(speciesId, selectedOrganization.id.toString());
```

Comment:
do we have enough info to reinterpret the search result payload as an api payload? just curious about this fetch..

Reply:
It's theoretically possible but would be awkward with the way the code is set up right now. You'd have to map from the previous locale's display names for the enum fields to the new locale's names, but the way it's structured right now, by the time you know the new names, the old ones have already been discarded from the `strings` object, so you'd have to keep a cache around.

If we care about this, a better approach might be to stop having the search API expect localized enum values as input. I thought about doing that but I didn't like that it would mean you'd do "search for the exact value `Foo`" and get back results where the value is `Bar` (or whatever the translation of `Foo` is). The way it is now is inconsistent too, though, just in a different way.

Reply:
Ok, got it, makes total sense.
Avoiding a network request between user click and viewing the modal would be ideal but in this case correctness is more important.